### PR TITLE
Migrate Embedder API documentation to Doxygen format.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Fixes https://github.com/flutter/flutter/issues/38870. This makes generated documentation a whole lot nicer and more navigable.

### CODE REVIEW

File: shell/platform/embedder/embedder.h [hunk 1]
```diff
@@ -41,205 +41,208 @@ typedef enum {
   kSoftware,
 } FlutterRendererType;
 
-// Additional accessibility features that may be enabled by the platform.
```
Comment (stuartmorgan) [inline shell/platform/embedder/embedder.h]: Was this line supposed to be removed? Seems like it was useful.

File: shell/platform/embedder/embedder.h [hunk 2]
```diff
@@ -434,103 +438,104 @@ typedef struct {
   double bottom;
 } FlutterRect;
 
-// |FlutterSemanticsNode| ID used as a sentinel to signal the end of a batch of
-// semantics node updates.
+/// |FlutterSemanticsNode| ID used as a sentinel to signal the end of a batch of
```
Comment (stuartmorgan) [inline shell/platform/embedder/embedder.h]: s/|/`/ ?

File: shell/platform/embedder/embedder.h [hunk 3]
```diff
@@ -434,103 +438,104 @@ typedef struct {
...
-// The semantics tree is maintained during the semantics phase of the pipeline
-// (i.e., during PipelineOwner.flushSemantics), which happens after
-// compositing. Updates are then pushed to embedders via the registered
-// |FlutterUpdateSemanticsNodeCallback|.
+/// A node that represents some semantic data.
+///
+/// The semantics tree is maintained during the semantics phase of the pipeline
+/// (i.e., during PipelineOwner.flushSemantics), which happens after
+/// compositing. Updates are then pushed to embedders via the registered
+/// |FlutterUpdateSemanticsNodeCallback|.
```
Comment (stuartmorgan) [inline shell/platform/embedder/embedder.h]: Same. Search for all |'s?

File: shell/platform/embedder/embedder.h [hunk 4]
```diff
@@ -944,85 +957,195 @@ FlutterEngineResult FlutterEngineSendPlatformMessage(
...
 
-// This API is only meant to be used by platforms that need to flush tasks on a
-// message loop not controlled by the Flutter engine. This API will be
-// deprecated soon.
+//------------------------------------------------------------------------------
+/// @brief      This API is only meant to be used by platforms that need to
+///             flush tasks on a message loop not controlled by the Flutter
+///             engine.
+///
+/// @attention  This API will be deprecated and is not part of the stable API.
```
Comment (stuartmorgan) [inline shell/platform/embedder/embedder.h]: You use `@deprecated` elsewhere; why not here?

File: shell/platform/embedder/embedder.h [hunk 5]
```diff
@@ -944,85 +957,195 @@ FlutterEngineResult FlutterEngineSendPlatformMessage(
...
+/// @see        FlutterEngineUnregisterExternalTexture()
+/// @see        FlutterEngineMarkExternalTextureFrameAvailable()
+///
+/// @param[in]  engine              A running engine instance.
+/// @param[in]  texture_identifier  The identifier of the texture to register
+///                                 with the engine. The embedder may supply new
+///                                 frames to this texture using the same
+///                                 identifier.
+///
+/// @return     { description_of_the_return_value }
```
Comment (stuartmorgan) [inline shell/platform/embedder/embedder.h]: ^^^

File: shell/platform/embedder/embedder.h (hunk 4 above)
Comment (chinmaygarde) [reply shell/platform/embedder/embedder.h]: Fixed.

File: shell/platform/embedder/embedder.h (hunk 2 above)
Comment (chinmaygarde) [reply shell/platform/embedder/embedder.h]: Done. Here and elsewhere.

File: shell/platform/embedder/embedder.h (hunk 3 above)
Comment (chinmaygarde) [reply shell/platform/embedder/embedder.h]: Done.

File: shell/platform/embedder/embedder.h (hunk 5 above)
Comment (chinmaygarde) [reply shell/platform/embedder/embedder.h]: Done.

File: shell/platform/embedder/embedder.h (hunk 1 above)
Comment (chinmaygarde) [reply shell/platform/embedder/embedder.h]: No. Fixed.

**Comment by chinmaygarde** [comment]
LUCI is red because of goma infra failures. Pushing this poke at the bots.

**Pull request merged**